memgpt/server/server.py

```diff
@@ -1235,7 +1240,9 @@ def get_agent_state(self, user_id: str, agent_id: Optional[str], agent_name: Opt
 
         # Get the agent object (loaded in memory)
         memgpt_agent = self._get_or_load_agent(agent_id=agent_id)
-        return memgpt_agent.agent_state
+        assert isinstance(memgpt_agent.memory, Memory)
+        assert isinstance(memgpt_agent.agent_state.memory, Memory)
+        return memgpt_agent.agent_state.model_copy(deep=True)
```

Comment:
@sarahwooders this `model_copy(deep=True)` is presumably what fixed the bugs in `test_new_client`...